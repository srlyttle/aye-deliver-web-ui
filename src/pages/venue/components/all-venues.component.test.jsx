import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';
import AllVenues from './all-venues.component';
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
});
function renderWithRouter(
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}

test('<AllVenues is rendered correctly', async () => {
  const confirmVenueDelete = () => {};
  const allVenueData = [
    {
      createdAt: '2020-06-02T15:36:04.492Z',
      id: '58ae3921-df19-4427-a183-50174ea47e94',
      name: 'Hello world!',
      owner: 'cf950200-670b-4dda-adca-d022f085d266',
      products: { nextToken: null },
      tags: ['Cafe', 'Restaurant', 'Groceries', 'Alchohol'],
      updatedAt: '2020-06-02T15:36:04.492Z'
    }
  ];
  const allVenues = (
    <AllVenues
      tableData={allVenueData}
      confirmVenueDelete={confirmVenueDelete}
    />
  );

  const { findByText, container } = renderWithRouter(allVenues);

  const item = await findByText(allVenueData[0].name);
  expect(item.innerHTML).toBe(allVenueData[0].name);
  expect(container).toMatchSnapshot();
  //ant-tag ant-tag-blue
  expect(
    document.querySelectorAll('tbody tr:first-child .ant-tag').length
  ).toBe(4);
  const editLink = document.querySelector('tbody tr:first-child a');
  expect(editLink.getAttribute('href')).toBe(`/venue/${allVenueData[0].id}`);
});
