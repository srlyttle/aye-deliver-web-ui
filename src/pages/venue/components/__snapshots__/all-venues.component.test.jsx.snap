// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllVenues is rendered correctly 1`] = `
<div>
  <div
    class="sub-title"
  />
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      Name
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Tags
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Action
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="58ae3921-df19-4427-a183-50174ea47e94"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      Hello world!
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <span
                        class="ant-tag ant-tag-blue"
                      >
                        Cafe
                      </span>
                      <span
                        class="ant-tag ant-tag-blue"
                      >
                        Restaurant
                      </span>
                      <span
                        class="ant-tag ant-tag-blue"
                      >
                        Groceries
                      </span>
                      <span
                        class="ant-tag ant-tag-blue"
                      >
                        Alchohol
                      </span>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="ant-space ant-space-horizontal ant-space-align-center"
                      >
                        <div
                          class="ant-space-item"
                          style="margin-right: 16px;"
                        >
                          <a
                            class="sidedrawer_item_link"
                            href="/venue/58ae3921-df19-4427-a183-50174ea47e94"
                          >
                            Details
                          </a>
                        </div>
                        <div
                          class="ant-space-item"
                        >
                          <span>
                            Delete
                          </span>
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination ant-table-pagination-right"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-prev ant-pagination-disabled"
            title="Previous Page"
          >
            <a
              class="ant-pagination-item-link"
              disabled=""
            >
              <span
                aria-label="left"
                class="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </a>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-next ant-pagination-disabled"
            title="Next Page"
          >
            <a
              class="ant-pagination-item-link"
              disabled=""
            >
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
